src/settings/window_geometry.rs

```diff
@@ -1,16 +1,73 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
-pub const DEFAULT_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
+pub fn try_to_load_last_window_size() -> Result<WindowGeometry, String> {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(".local/share/nvim/neovide-settings.txt");
+    let serialized_size = std::fs::read_to_string(&settings_path).map_err(|e| e.to_string())?;
+
+    let de: PhysicalSize<u32> =
+        serde_json::from_str(&serialized_size).map_err(|e| e.to_string())?;
+    log::debug!("Loaded Window Size: {:?}", de);
+    Ok(WindowGeometry {
+        width: de.width as u64,
+        height: de.height as u64,
+    })
+}
+#[cfg(unix)]
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(".local/share/nvim/neovide-settings.txt");
+    settings_path
+}
+#[cfg(windows)]
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push("AppData/Local/nvim-data/neovide-settings.txt");
+    settings_path
+}
+
+const PRIVATE_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
```

Comment:
I think it's better to leave this as `DEFAULT_WINDOW_GEOMETRY`, and make it private.

And instead of lazy static, make function `get_default_window_geometry`, and use it in settings.

Reply:
Yeah sounds good. I think I chose to do that because previously the parsing window geometry function was called in multiple places in the codebase.